src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -68,6 +69,7 @@ public class ScriptHost : JobHost
         private IFunctionRegistry _functionDispatcher;
         private ILoggerFactory _loggerFactory;
         private IProcessRegistry _processRegistry = new EmptyProcessRegistry();
+        private IJobHostMetadataProvider _metadataProvider;
```

Comment:
Why does this need to be a field? Because the correct time for its use is order dependent on the initialization pipeline, I think you should just leave it a local variable where it is used

Reply:
Removed